src/components/calcite-radio-button/calcite-radio-button.tsx

```diff
@@ -129,20 +129,22 @@ export class CalciteRadioButton {
   //
   //--------------------------------------------------------------------------
 
-  private checkFirstRadioButton(): void {
+  private checkLastRadioButton(): void {
     const radioButtons = Array.from(document.querySelectorAll("calcite-radio-button")).filter(
       (radioButton) => radioButton.name === this.name
     ) as HTMLCalciteRadioButtonElement[];
-    let firstCheckedRadioButton: HTMLCalciteRadioButtonElement;
-    if (radioButtons?.length > 0) {
-      radioButtons.forEach((radioButton: HTMLCalciteRadioButtonElement) => {
-        if (firstCheckedRadioButton) {
-          radioButton.checked = false;
-          this.calciteRadioButtonCheckedChange.emit(false);
-        } else if (radioButton.checked) {
-          firstCheckedRadioButton = radioButton;
-        }
-      });
+
+    const checkedRadioButtons = radioButtons?.filter(
+      (radioButton) => radioButton.checked
+    ) as HTMLCalciteRadioButtonElement[];
+
+    if (radioButtons?.length > 0 && checkedRadioButtons?.length > 1) {
```

Comment:
I think you only need to check `checkedRadioButtons.length > 1`  since it implicitly means that `radioButtons.length > 0`.